src/core/pointcloud/qgspointcloudindex.cpp

```diff
@@ -250,7 +250,7 @@ int QgsPointCloudIndex::nodePointCount( const IndexedPointCloudNode &n )
   int count = -1;
```

Comment:
```suggestion
  mHierarchyMutex.lock();
  const int count = mHierarchy.value( n, -1 );
  mHierarchyMutex.unlock();
  return count;
```

That avoids the double hash lookup (contains and then [ ] )